Gruntfile.js

```diff
@@ -151,77 +160,102 @@ module.exports = (grunt) => {
+      buildDebugNoFonts: debugNoFonts,
+
+      watchProdAllFonts: { ...prodAllFonts, ...watch },
+      watchProdNoFonts: { ...prodNoFonts, ...watch },
+      watchProdBravuraOnly: { ...prodBravuraOnly, ...watch },
+      watchProdGonvilleOnly: { ...prodGonvilleOnly, ...watch },
+      watchProdPetalumaOnly: { ...prodPetalumaOnly, ...watch },
+
+      watchDebug: { ...debugAllFonts, ...watch },
+      watchDebugPlusTests: { ...debugAllFontsWithTests, ...watch },
+      watchDebugNoFonts: { ...debugNoFonts, ...watch },
     },
     concurrent: {
       options: {
         logConcurrentOutput: true,
-        indent: false,
+        indent: true,
       },
-      dev: ['webpack:watchDev', 'webpack:watchTest'],
-      static: ['webpack:watchStatic'],
-      dynamic: ['webpack:watchDynamic'],
-      staticAndDynamic: ['webpack:watchStatic', 'webpack:watchDynamic'],
+      debug: [
+        // While developing, you can speed up builds by temporarily commenting out one of the tasks below.
+        'webpack:watchDebug',
+        'webpack:watchDebugPlusTests',
+      ],
+      production: [
+        // While developing, you can speed up builds by temporarily commenting out one of the tasks below.
+        'webpack:watchProdAllFonts',
+        'webpack:watchProdNoFonts',
+      ],
     },
+
     eslint: {
-      target: SOURCES.concat('./tests'),
+      target: ['./src', './tests'],
       options: { fix: true },
     },
     qunit: {
       files: ['tests/flow-headless-browser.html'],
     },
     copy: {
-      // copy the vexflow.js production build to vexflow.module.js
-      module: {
-        src: path.join(BUILD_DIR, 'vexflow.js'),
-        dest: path.join(BUILD_DIR, 'vexflow.module.js'),
+      modulePackageJSON: {
+        expand: true,
+        cwd: BASE_DIR,
+        src: ['tools/esm/package.json'],
+        dest: path.join(BUILD_DIR, 'esm/'),
+        flatten: true,
+      },
+      // Modules: For now, we just copy the vexflow****.js to the esm/ folder and add exports at the bottom.
+      moduleJSFiles: {
+        expand: true,
+        cwd: BUILD_DIR,
+        src: ['*.js'],
+        dest: path.join(BUILD_DIR, 'esm/'),
+        flatten: true,
         options: {
-          process: function (content, srcpath) {
-            // Insert the export line BEFORE the source map comment.
-            const srcMapLinePrefix = '//# sourceMappingURL';
-            const exportVex = 'export default Vex;\n';
-            return content.replace(srcMapLinePrefix, exportVex + srcMapLinePrefix);
+          process: (content) => {
+            // We take a JS file that sets globalThis.Vex, and turn it into a ES module.
+            // To do this, we extract all the classes from Vex.Flow, and re-export them.
+            // We also export Vex both as a named export and as a default export.
+            // THIS IS A HACK, so we should figure out how to use webpack to export both ESM and CJS.
+            const exports = `
```

Comment:
Not thrilled about this -- can we load this in dynamically from a separate file in the root directory? i.e.,

- put the contents in a file and load it in this `Gruntfile`.
- update dev instructions to call this out as a step
- add a bug to remove this hack (or find a suitable mitigation)

Reply:
Alternatively, can we drop support for CJS?

Reply:
I discussed this here: https://github.com/0xfe/vexflow/issues/1222

I have a PR coming for native ESM support that will remove this hack. Hopefully this will be in 4.0 beta 2.

Not sure when you'll want to drop support for CJS. It's the classic old reliable script tag that every web developer learns as a hello world. My proposal is to keep everything in vexflow/build/, and have a `build/esm` folder for the modern module files, and a `build/cjs/` for the pre-bundled files compatible with script tags.

Reply:
We also have quite a few hobby developers still using plain JS/HTML with script tags in OSMD. It would be kinda nice to keep CJS support. I believe ronyeh is working on making CJS a different build target as well, so it shouldn't negatively affect the esm build too much.

Reply:
Okay, thanks yeah, I agree with that.